### ISSUE TITLE
`no-inferred-empty-object-type` should throw an error when calling a method

### ISSUE THREAD

**Issue by clebert** [palantir/tslint#2908]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
type Script<T> = (callback: (result: T) => void) => void;

class Browser {
  public async execute<T>(script: Script<T>): Promise<T> {
    return;
  }
}

const browser = new Browser();
browser.execute(callback => callback('foo'));
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-inferred-empty-object-type": true
  }
}
```

#### Actual behavior

No warning/error

#### Expected behavior

Error: Explicit type parameter needs to be provided to the method call

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!
